### DISCUSSION TITLE
[item 8] 스레드 제어

### DISCUSSION THREAD

**Discussion by yerim123456** [GDSC-SWU/2023-JAVA-Study#151] [Q&A] [answered]
41p에 있는 _cleaner는 자신을 수행할 스레드를 제어할 수 있다는 면에서 조금 낫다._ 라는 문장의 스레드를 제어한다는 의미가 궁금합니다.

**Comment by nusuy** [comment, marked as answer]
스레드를 제어한다는 것의 의미는 스레드의 상태를 변경시킬 수 있다는 의미이고, 예로는 `interrupt()`, `sleep()` 등이 있습니다.

하지만 이는 스레드 간의 우선순위를 제어할 수 있다는 의미는 아닙니다.

**Reply by hyojeongchoi** [reply]
interrup는 실행 대기열로 스레드를 이동시키지만 앞선 스레드들이 실행되고난 후에야 실행이 됩니다.
